analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ConditionEvaluatesToConstantBase.cs

```diff
@@ -94,61 +95,67 @@ public override void ExecutionCompleted()
     private void ReportIssue(IOperation operation, BasicBlock block, bool conditionValue)
     {
         var issueMessage = operation.Kind == OperationKindEx.IsNull ? MessageNull : string.Format(MessageBool, conditionValue);
-        var secondaryLocations = SecondaryLocations(block, conditionValue);
+        // For SwitchExpressionArms like `true => 5` we are only interested in the left part (`true`).
+        var syntax = operation.Syntax.IsKind(SyntaxKindEx.SwitchExpressionArm) ? ((SwitchExpressionArmSyntaxWrapper)operation.Syntax).Pattern : operation.Syntax;
+        var secondaryLocations = SecondaryLocations(block, conditionValue, syntax);
         if (secondaryLocations.Any())
         {
-            ReportIssue(Rule2583, operation, secondaryLocations, issueMessage, S2583MessageSuffix);
+            ReportIssue(Rule2583, syntax, secondaryLocations, issueMessage, S2583MessageSuffix);
         }
         else
         {
-            ReportIssue(Rule2589, operation, null, issueMessage, string.Empty);
+            ReportIssue(Rule2589, syntax, null, issueMessage, string.Empty);
         }
     }
 
-    private List<Location> SecondaryLocations(BasicBlock block, bool conditionValue)
+    private List<Location> SecondaryLocations(BasicBlock block, bool conditionValue, SyntaxNode conditionSyntax)
     {
         List<Location> locations = new();
-        IOperation currentStart = null;
-        IOperation currentEnd = null;
-        var unreachable = UnreachableOperations(block, conditionValue).Where(x => !IsIgnoredLocation(x.Syntax)).ToHashSet();
-        foreach (var operation in unreachable.Concat(reached).OrderBy(x => x.Syntax.SpanStart))
+        var unreachable = UnreachableOperations(block, conditionValue);
+        var currentStart = conditionSyntax.Span.End;
```

Comment:
The `conditionSyntax.Span.End` is here, and then in the lambda on the next row. Reusing the variable will make it more readable and tiny little more performant. 
You can also consider making it a parameter and do the `.Span.End` on the invocation side